src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random random = new Random();
+    private static final CircleSupplier circle = new CircleSupplier();

```

Comment:
The 'circle', 'isoscelesTrapezoid', 'rectangle', 'square', and 'rightTriangle' objects should be declared as 'final', as their values should not change.

Reply:
But they should be randomised in each row. So they will be changing.